test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -301,6 +315,66 @@ func runFuncWithRecover(f pulumi.RunFunc) pulumi.RunFunc {
 	}
 }
 
+func shouldRetryError(err error) bool {
+	// Add here errors that are known to be flakes and that should be retried
+	if strings.Contains(err.Error(), "i/o timeout") {
+		return true
+	}
+
+	if strings.Contains(err.Error(), "creating EC2 Instance: IdempotentParameterMismatch:") {
+		return true
+	}
+
+	if strings.Contains(err.Error(), "InvalidInstanceID.NotFound") {
+		return true
+	}
+
+	return false
+}
+
+// sendEventToDatadog sends an event to Datadog, it will use the API Key from environment variable DD_API_KEY if present, otherwise it will use the one from SSM Parameter Store
+func sendEventToDatadog(title string, message string, tags []string, logger io.Writer) error {
+
+	apiKey, err := runner.GetProfile().ParamStore().GetWithDefault(parameters.APIKey, "")
+	if err != nil {
+		fmt.Fprintf(logger, "error when getting API key from parameter store: %v", err)
+		return err
+	}
+
+	apiKeyFromEnv, ok := os.LookupEnv("DD_API_KEY")
```

Comment:
That's not required. The profiles and param stores are handling that for you. Locally the key will be read from config file `~/.test_infra_config.yaml` (which normally already has an APIKey) or from `E2E_<param_name>`.

Reply:
That's good to know! Removed